common/beerocks/bcl/include/bcl/beerocks_promise.h

```diff
@@ -22,53 +23,70 @@ template <class T> class promise {
     // Set value for waiting threads
     void set_value(const T value)
     {
-        pthread_mutex_lock(&m_mut);
+        int err;
+        if ((err = pthread_mutex_lock(&m_mut)) != 0) {
+            LOG(ERROR) << "pthread_mutex_lock failed, error code: " << err;
+            return;
+        }
```

Comment:
Possibly not for this PR, but this really should be an assert. The caller doesn't check the return value of set_value (because there isn't any), but does continue life as if it was set successfully.

The same comment applies to all changes in this commit.